nilearn/version.py

```diff
@@ -1 +1,91 @@
+# *- encoding: utf-8 -*-
+"""
+nilearn version, required package versions, and utilities for checking
+"""
+# Author: Alexandre Abraham, Philippe Gervais, Ben Cipollini
+# License: simplified BSD
+
 __version__ = '0.1b1'
+
+# All metadata needed for required modules:
+#    * minver: minimum required version of the module
+#    * manual: whether the module must be installed manually
+# See more info in check_module_dependencies docstring.
+#
+# This is a list to preserve order, and to avoid the Python 2.7
+#   dependency of collections.OrderedDict
+REQUIRED_MODULE_METADATA = (
+    ('numpy', {
+        'minver': '1.6.0',
+        'manual_install': True,
+        'install_info': 'Install via your OS package manager (Linux) or Anaconda (OSX, Windows).'}),
+    ('scipy', {
+        'minver': '0.9.0',
+        'manual_install': True,
+        'install_info': 'Install via your OS package manager (Linux) or Anaconda (OSX, Windows).'}),
+    ('sklearn', {
+        'minver': '0.10',
+        'manual_install': True,
+        'install_info': 'Install via your OS package manager (Linux) or Anaconda (OSX, Windows).'}),
+    ('nibabel', {
+        'minver': '1.1.0',
+        'manual_install': False}),
+    ('gzip', {
+        'minver': '0.0.0',
+        'manual_install': True,
+        'install_info': 'Use a python version compiled with gzip.'}))
+
+
+def _import_module_with_version_check(module_name, minimum_version, install_info=None):
+    """Check that module is installed with a recent enough version
+    """
+    from distutils.version import LooseVersion
+
+    try:
+        module = __import__(module_name)
+    except ImportError as exc:
+        user_friendly_info = ('Module "{}" could not be found.  {}').format(
+            module_name,
+            install_info or 'Please install it properly to use nilearn.')
+        exc.args += (user_friendly_info,)
+        raise
+
+    # Avoid choking on modules with no __version__ attribute
+    module_version = getattr(module, '__version__', '0.0.0')
+
+    version_too_old = (not LooseVersion(module_version) >=
+                       LooseVersion(minimum_version))
+
+    if version_too_old:
+        message = (
+            'A {module_name} version of at least {minimum_version} '
+            'is required to use nilearn. {module_version} was found. '
+            'Please upgrade {module_name}').format(
+                module_name=module_name,
+                minimum_version=minimum_version,
+                module_version=module_version)
+
+        raise ImportError(message)
+
+    return module
+
+def _check_module_dependencies(manual_install_only=False):
```

Comment:
not a big fan of the `manual_install` naming but I have to admit I haven't found a great alternative yet ...